runDocker.ts

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Dockerode, { Container } from 'dockerode'
+import { fork } from 'node:child_process'
+import { LauncherFunction } from './run.js'
+
+const [, , command, jsonifiedArgs] = process.argv
+const docker = new Dockerode({ protocol: 'http' })
+
+let containerId = ''
+
+if (command === 'launch-ddb-local-docker-subprocess') {
+  const { dataDir, logsDir, port, options } = JSON.parse(jsonifiedArgs)
+
+  const container = await docker.createContainer({
+    AttachStderr:true,
+    AttachStdin:true,
+    AttachStdout:true,
+    Image: 'amazon/dynamodb-local:latest',
+    name: 'dynamodb-local',
+    Tty: true,
+    Cmd: [
+      '-jar',
+      'DynamoDBLocal.jar',
+      '-sharedDb',
+      '-dbPath',
+      '/home/dynamodblocal/data',
+    ],
+    WorkingDir: '/home/dynamodblocal',
+    ExposedPorts: {
+      '8000/tcp': {},
+    },
+    Env: [
+      ...options,
+      'path.data=/var/lib/ddb-local',
+      'path.logs=/var/log/ddb-local',
+    ],
+    HostConfig: {
+      Binds: [`${dataDir}:/home/dynamodblocal/data`],
```

Comment:
Remove bind mount. There's no need to persist the data to the host filesystem.

Reply:
I realized i could just use the run function from docker, I have updated it in the latest commit. I still need to clean it up, but it should be behaving as expected.